indico/web/client/js/react/components/PrincipalSearchField.jsx

```diff
@@ -48,61 +49,88 @@ const searchGroups = async (data) => {
         return;
     }
 
-    return response.data;
+    return camelizeKeys(response.data);
 };
 
 export default class PrincipalSearchField extends React.Component {
     static propTypes = {
         multiple: PropTypes.bool,
-        defaultValue: PropTypes.object,
+        value: PropTypes.oneOfType([
+            PropTypes.object,
+            PropTypes.array
+        ]),
         onChange: PropTypes.func.isRequired,
         disabled: PropTypes.bool,
         withGroups: PropTypes.bool
     };
 
     static defaultProps = {
         multiple: false,
-        defaultValue: null,
+        value: null,
         disabled: false,
         withGroups: false
     };
 
     constructor(props) {
         super(props);
-        const {defaultValue} = this.props;
-        const userId = defaultValue && defaultValue.id;
+
+        const {multiple, onChange} = this.props;
+        let {value} = this.props;
+        let dropdownValue;
+
+        if (multiple) {
+            value = value || [];
+            dropdownValue = value.map((val) => val.identifier);
+        } else {
+            dropdownValue = value && value.identifier;
+        }
 
         this.state = {
             isFetching: false,
             options: [],
-            prevPropsUserId: userId,
-            userId
+            prevPropsValue: dropdownValue,
+            value: dropdownValue
         };
-        this.userCache = {};
+
+        if (multiple) {
+            this.userCache = Object.assign({}, ...value.map((val) => ({[val.identifier]: val})));
+        } else {
+            this.userCache = {};
+        }
+
+        onChange(value);
```

Comment:
@mic4ael do you remember why these calls to the onChange callback are/were needed?

Reply:
yeah, I think in this case it is needed when editing a booking since it sets the initial value on the instance of the form which allows to properly perform later validations